compare_hashes/score.h

```diff
@@ -0,0 +1,60 @@
+#ifndef SCORE_H
+#define SCORE_H
+
+#include <cassert>
+
+typedef int location_in_submission;
+typedef unsigned int hash;
+typedef std::string user_id;
+typedef unsigned int version_number;
+
+// represents the plagiarism score for a given submissions, used for the overall rankings file
+class Score {
+public:
+  // CONSTRUCTOR
+  Score(unsigned int hashes_matched, float percent): hashes_matched(hashes_matched), percent(percent), score(-1) {}
+  Score(const Score &other) { copy(other); }
+
+  // GETTERS
+  float getPercent() const { return percent; }
+  unsigned int getHashesMatched() const { return hashes_matched; }
+
+  // MODIFIER
+  // Each submission in the ranking file gets a composite score that weighs both its percentage
+  // of suspicious matches, and its percentile of total number of hashes matched
+  void calculateScore(unsigned int max_hashes_matched) {
+    assert(PERCENT_WEIGHT + MATCH_WEIGHT == 1);

```

Comment:
If the values of PERCENT WEIGHT or MATCH WEIGHT never change why is this assert needed? Disregard this comment if that will change in the future

Reply:
It's done more as a sanity check than anything.  The two constants may be adjusted in the future, or new ones may be added, and we want to make sure that this code isn't forgotten when those values are changed.  Suggestions on how to do it in a way such that it causes a compilation error instead of a runtime error?

Reply:
You can use static assert for compile time constexpr checks

```c++
static_assert(PERCENT_WEIGHT + MATCH_WEIGHT == 1, "Weights don't add up 1!");
```

Or if you'd like to check outside the class
```c++
static_assert(Score::PERCENT_WEIGHT + Score::MATCH_WEIGHT == 1, "Weights don't add up 1!");
```

Reply:
Oh, ok, nice.  I'll switch it to use that syntax in my next changes.  That's way way better than checking every time getScore() is calculated.